pkg/controllers/dynakube/controller.go

```diff
@@ -261,14 +261,14 @@ func (controller *Controller) reconcileDynaKube(ctx context.Context, dynakube *d
 
 	log.Info("start reconciling process module config")
 
-	err = processmoduleconfigsecret.NewReconciler(
-		controller.client, controller.apiReader, dynatraceClient, dynakube, controller.scheme, timeprovider.New()).
-		Reconcile(ctx)
+	err = controller.reconcileComponents(ctx, dynatraceClient, istioClient, dynakube)
 	if err != nil {
 		return err
 	}
 
-	return controller.reconcileComponents(ctx, dynatraceClient, istioClient, dynakube)
+	return processmoduleconfigsecret.NewReconciler(
+		controller.client, controller.apiReader, dynatraceClient, dynakube, controller.scheme, timeprovider.New()).
+		Reconcile(ctx)
```

Comment:
It used to be here and for the refactoring to work I needed to move reconcilers because the oneagent tenant token needs to be created in order to set oneagent connection info. 

But @0sewa0 also commented and I might move processmoduleconfig reconciler into the injection reconciler. I am currently looking into it

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/2777/commits/21e52638bf7c80d7a47af88adb0f0875282826bc